xray/backends/pynio_.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+from .. import Variable
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin
+from ..core import indexing
+
+from .common import AbstractDataStore
+
+
+class NioArrayWrapper(NDArrayMixin):
+    def __init__(self, array, ds):
+        self.array = array
+        self._ds = ds  # make an explicit reference because pynio uses weakrefs
```

Comment:
Fair enough -- there are some tricky tradeoffs here. I know Jeff Whitaker has dealt with similar issues in the Python netCDF4 library. In any case I think this work around covers things well enough for xray's purposes.